build/dnsmasq/build.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/bash
+#
+# {{{ CDDL HEADER
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+# }}}
+
+# Copyright 2022 OmniOS Community Edition (OmniOSce) Association.
+
+. ../../lib/build.sh
+
+PROG=dnsmasq
+VER=2.86
+PKG=ooce/network/dnsmasq
+SUMMARY="Lightweight, easy to configure DNS forwarder"
+DESC="dnsmasq is a lightweight, easy to configure DNS forwarder, designed to "
+DESC+="provide DNS (and optionally DHCP and TFTP) services to a small-scale network."
+
+set_arch 64
+set_mirror 'https://thekelleys.org.uk/'
+set_checksum sha256 '28d52cfc9e2004ac4f85274f52b32e1647b4dbc9761b82e7de1e41c49907eb08'
+
+BASEDIR=$PREFIX/$PROG
+CONFFILE=/etc$BASEDIR/$PROG.conf
+EXECFILE=$PREFIX/sbin/dnsmasq
+
+copy_sample_config() {
+    local relative_conffile=${CONFFILE#/}
+    local dest_confdir=$DESTDIR/${relative_conffile%/*}
+
+    logmsg "-- copying sample config"
+    logcmd mkdir -p "$dest_confdir" || logerr "mkdir failed"
+    logcmd cp $TMPDIR/$BUILDDIR/$PROG.conf.example $DESTDIR/$relative_conffile \
+        || logerr "copying configs failed"
+}
+
+# No configure
+configure64() { :; }
+
+XFORM_ARGS="
+    -DPREFIX=${PREFIX#/}
+    -DBASEDIR=${BASEDIR#/}
+    -DEXECFILE=$EXECFILE
+    -DCONFFILE=$CONFFILE
+    -DUSER=$PROG
+    -DGROUP=$PROG
+    -DPROG=$PROG
+"
+
+MAKE_ARGS_WS="
+    CC=$CC
+    CFLAGS=\"-DNO_IPSET $CFLAGS $CFLAGS64 $CTF_CFLAGS\"
+    LDFLAGS=\"$LDFLAGS $LDFLAGS64\"
+    PREFIX=$PREFIX
+    MANDIR=$PREFIX/share/man
+    sunos_libs=\"-lnsl -lsocket\"
+"
+
+MAKE_INSTALL_ARGS_WS="
+    PREFIX=$PREFIX
+    MANDIR=$PREFIX/share/man
+"
+
+init
+download_source "$PROG" "$PROG" "$VER"
+patch_source
+prep_build
+build
+copy_sample_config
+xform files/$PROG.xml > $TMPDIR/$PROG.xml
+install_smf network $PROG.xml
```

Comment:
Can you please change to this? We're trying to adopt an approach of keeping the 'extra' manifests in a separate subdirectory. The service should still be `svc:/network/dns/dnsmasq:default` - no ooce there.
```suggestion
install_smf ooce $PROG.xml
```